### PR TITLE
Add support for spot instances

### PR DESCRIPTION
**Pull request by alexander-demichev**
This PR is follow-up for https://github.com/openshift/cluster-api-provider-azure/pull/138 and adds support for spot instances.
API exposure of spot instances to provider spec and label setting was added in the previous PR.

https://github.com/openshift/enhancements/blob/master/enhancements/machine-api/spot-instances.md
https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/559

### CODE REVIEW

File: pkg/cloud/azure/actuators/machine/reconciler.go [hunk 1]
```diff
@@ -610,3 +618,21 @@ func (s *Reconciler) getCustomUserData() (string, error) {
 
 	return base64.StdEncoding.EncodeToString(data), nil
 }
+
+func getSpotVMOptions(spotVMOptions *v1beta1.SpotVMOptions) (compute.VirtualMachinePriorityTypes, compute.VirtualMachineEvictionPolicyTypes, *compute.BillingProfile, error) {
```
Comment (enxebre) [inline pkg/cloud/azure/actuators/machine/reconciler.go:623]: Can we add a unit test for this function?

File: pkg/cloud/azure/actuators/machine/reconciler.go [hunk 2]
```diff
@@ -610,3 +618,21 @@ func (s *Reconciler) getCustomUserData() (string, error) {
...
+	if spotVMOptions.MaxPrice != nil {
+		maxPrice, err := strconv.ParseFloat(*spotVMOptions.MaxPrice, 64)
+		if err != nil {
+			return compute.VirtualMachinePriorityTypes(""), compute.VirtualMachineEvictionPolicyTypes(""), nil, err
+		}
+		billingProfile = &compute.BillingProfile{
+			MaxPrice: &maxPrice,
+		}
+	}
+	return compute.Spot, compute.Deallocate, billingProfile, nil
```
Comment (enxebre) [inline pkg/cloud/azure/actuators/machine/reconciler.go:638]: Can we add a comment elaborating why you chose `Deallocate`?

**Comment by enxebre** [comment]
Can we add a ref in the commit desc to https://github.com/openshift/cluster-api-provider-azure/pull/138 and explain support for spot instances was split into that PR and this one?

**Comment by alexander-demichev** [comment]
@enxebre all done

**Pull request merged**